cg/apps/lims/api.py

```diff
@@ -128,35 +134,31 @@ def get_delivery_date(self, lims_id: str) -> dt.date:
             date = None
         return date
 
-    def capture_kit(self, lims_id: str) -> str:
+    def capture_kit(self, lims_id: str) -> str | None:
         """Get capture kit for a LIMS sample."""
-
         step_names_udfs = MASTER_STEPS_UDFS["capture_kit_step"]
         capture_kits = set()
-
-        lims_sample = Sample(self, id=lims_id)
-        capture_kit = lims_sample.udf.get("Bait Set")
-
-        if capture_kit and capture_kit != "NA":
-            return capture_kit
-
-        for process_type in step_names_udfs:
-            artifacts = self.get_artifacts(
-                samplelimsid=lims_id, process_type=process_type, type="Analyte"
-            )
-            udf_key = step_names_udfs[process_type]
-            capture_kits = capture_kits.union(
-                self._find_capture_kits(artifacts, udf_key)
-                or self._find_twist_capture_kits(artifacts, udf_key)
-            )
-
-        if len(capture_kits) > 1:
-            message = f"Capture kit error: {lims_sample.id} | {capture_kits}"
-            raise LimsDataError(message)
-
-        if len(capture_kits) == 1:
-            return capture_kits.pop()
-
+        try:
+            lims_sample = Sample(self, id=lims_id)
+            capture_kit = lims_sample.udf.get("Bait Set")
```

Comment:
Would be great with a constant class instead of hard coded keys.
Add type hint.

Reply:
https://github.com/Clinical-Genomics/cg/issues/3163